rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java

```diff
@@ -174,21 +180,68 @@ void logUpdateCollections() {
         assertEvent(eventLogs, 0, BridgeEvents.UPDATE_COLLECTIONS.getEvent(), new Object[]{}, new Object[]{tx.getSender(signatureCache).toString()});
     }
 
-    @Test
-    void logAddSignature() {
-        // Setup logAddSignature params
-        BtcECKey federatorPubKey = BtcECKey.fromPrivate(BigInteger.valueOf(2L));
+    private static Stream<Arguments> logAddSignatureArgProvider() {
+        ActivationConfig.ForBlock fingerrootActivations = ActivationConfigsForTest.fingerroot500().forBlock(0);
+        ActivationConfig.ForBlock arrowHeadActivations = ActivationConfigsForTest.arrowhead600().forBlock(0);
+
+        FederationMember multiKeyFedMember = new FederationMember(
+            BtcECKey.fromPrivate(BigInteger.valueOf(100)),
+            ECKey.fromPrivate(BigInteger.valueOf(200)),
+            ECKey.fromPrivate(BigInteger.valueOf(300))
+        );
+        ECKey rskKeyFromMultiFedMemberBtcKey = ECKey.fromPublicOnly(multiKeyFedMember.getBtcPublicKey().getPubKey());
+        String wrongDerivedRskAddressFromMultiKeyFedMember = ByteUtil.toHexString(rskKeyFromMultiFedMemberBtcKey.getAddress());
+        String derivedRskAddressFromMultiKeyFedMember = ByteUtil.toHexString(multiKeyFedMember.getRskPublicKey().getAddress());
+
+        Arguments before_arrowhead_should_log_wrong_rskAddress_for_multiKey_FedMember = Arguments.of(fingerrootActivations, multiKeyFedMember, wrongDerivedRskAddressFromMultiKeyFedMember);
+        Arguments after_arrowhead_should_log_correct_rskAddress_for_multiKey_FedMember = Arguments.of(arrowHeadActivations, multiKeyFedMember, derivedRskAddressFromMultiKeyFedMember);
+
+        FederationMember singleKeyFedMember = new FederationMember(
+            BtcECKey.fromPrivate(BigInteger.valueOf(100)),
+            ECKey.fromPrivate(BigInteger.valueOf(100)),
+            ECKey.fromPrivate(BigInteger.valueOf(100))
+        );
+        ECKey rskKeyFromSingleFedMemberBtcKey = ECKey.fromPublicOnly(singleKeyFedMember.getBtcPublicKey().getPubKey());
+        String derivedRskAddressFromSingleBtcKeyFedMember = ByteUtil.toHexString(rskKeyFromSingleFedMemberBtcKey.getAddress());
+        String derivedRskAddressFromSingleRskKeyFedMember = ByteUtil.toHexString(singleKeyFedMember.getRskPublicKey().getAddress());
+
+        Arguments before_arrowhead_should_log_correct_rskAddress_for_single_FedMember = Arguments.of(fingerrootActivations, singleKeyFedMember, derivedRskAddressFromSingleBtcKeyFedMember);
+        Arguments after_arrowhead_should_log_correct_rskAddress_for_single_FedMember = Arguments.of(arrowHeadActivations, singleKeyFedMember, derivedRskAddressFromSingleRskKeyFedMember);
```

Comment:
I don't think this helps, you are creating test cases as arguments. I find it hard to follow.

It should be a lot simpler, pre and post arrowhead use one address or the other

Reply:
Fixed!